src/inmanta/agent/agent.py

```diff
@@ -1069,6 +1071,9 @@ async def update_agent_map(self, agent_map: dict[str, str]) -> None:
             await self._update_agent_map(agent_map)
 
     async def _update_agent_map(self, agent_map: dict[str, str]) -> None:
+        if "internal" not in agent_map:
+            raise ValueError("The internal agent must be present in the agent map")
+
```

Comment:
I'm not sure. The current behavior was to drop the internal agent altogether, so I don't think we have to worry too much about it being a breaking change (the previous behavior is just breaking in a different way). In that sense I think we can afford it, and if we can afford it I think it's the sensible thing to do. It also corresponds with the environment setting update behavior.

Reply:
@wouterdb what's your opinion on this?

Reply:
I think it is strange for the agent to reject the entire agent map, in case an entry is missing (that is can easily put in there itself).

I would understand if we reject the user input on the server side, here, I find it tricky, in that it can make existing, working setups stop working altogether. (this is code on the agent startup path)

Reply:
Ok, I'll change this.

Reply:
1. added a warning, I think this is definitely good.
2. auto-add it when it's missing. I could also just keep this part to the old behavior if you prefer.